backend/controllers/deckController.js

```diff
@@ -0,0 +1,349 @@
+
+    res.status(201).json({deck});
+  } catch (error) {
+    console.error('Error creating deck:', error);
+    res.status(500).json({error: 'Failed to create deck'});
+  }
+}
+
+/**
+ * @api {put} /decks/:deckId Edit an existing deck
+ * @apiName editDeck
+ * @apiGroup Deck
+ *
+ * @apiParam {String} deckId The ID of the deck to edit.
+ * @apiParam {String} name The new name of the deck.
+ * @apiParam {String} cover The new cover image of the deck.
+ * @apiParam {String} description The new description of the deck.
+ * @apiParam {Array} flashCards The new array of flash cards for the deck.
+ * @apiParam {Array} tags The new array of tags for the deck.
+ *
+ * @apiSuccess {Object} deck The edited deck object.
+ */
+async function editDeck(req, res) {
+  try {
+    const deckId = req.params.deckId;
+    const updatedFields = req.body;
+
+    const deck = await Deck.editDeck(deckId, updatedFields);
+
+    res.json({deck});
+  } catch (error) {
+    console.error('Error editing deck:', error);
+    res.status(500).json({error: 'Failed to edit deck'});
+  }
+}
+
+/**
+ * @api {post} /decks/:deckId/import Clone an existing deck
+ * @apiName cloneDeck
+ * @apiGroup Deck
+ *
+ * @apiParam {String} deckId The ID of the deck to clone.
+ *
+ * @apiSuccess (201) {Object} deck The cloned deck object.
+ */
+async function importDeck(req, res) {
+  try {
+    const deckId = req.params.deckId;
+    // const creatorId = req.user.id;
+
+    const creatorId = 123456;
+
+    const importedDeck = await Deck.importDeck(deckId, creatorId);
+
+    res.status(201).json({importedDeck});
+  } catch (error) {
+    console.error('Error cloning deck:', error);
+    res.status(500).json({error: 'Failed to clone deck'});
+  }
+}
+
+/**
+ * @api {post} /decks/:deckId/publish Publishes a deck in the marketplace
+ * @apiName publishDeck
+ * @apiGroup Deck
+ *
+ * @apiParam {String} deckId The ID of the deck to publish.
+ *
+ * @apiSuccess {Object} deck The published deck object.
+ */
+async function publishDeck(req, res) {
+  try {
+    const deckId = req.params.deckId;
+    const userId = req.user.id;
+
+    const deck = await Deck.publishDeck(deckId, userId);
+
+    res.json({deck});
+  } catch (error) {
+    console.error('Error publishing deck:', error);
+    res.status(500).json({error: 'Failed to publish deck'});
+  }
+}
+
+/**
+ * @api {post} /decks/:deckId/flashcards Appends a flashCard to a deck
+ * @apiName flashcards
+ * @apiGroup Deck
+ *
+ * Appends a flash card to a deck.
+ * @param {object} req - The request object.
+ * @param {object} res - The response object.
+ * @returns {Promise<void>} A Promise that resolves once the response is sent.
+ */
+async function appendFlashCardToDeck(req, res) {
+  try {
+    const deckId = req.params.deckId;
+    const {flashCard} = req.body;
+    const userId = req.user.id;
+
+    const deck = await Deck.findById(deckId);
+
+    if (!deck) {
+      return res.status(404).json({error: 'Deck not found'});
```

Comment:
Userid should defintely exist when it hits this piece of code, otherwise it should fail before reaching here. This functionality will be handled by the middleware